### ISSUE TITLE
study(rl): canonical transfer gates 与 λ dose-response coarse sweep

### ISSUE THREAD

**Issue by xcz0** [xcz0/Eco-AutoDrive#106]
Parent: #80

Depends on: #105

Blocks: #83

Historical prerequisites: #94, #98（closed / completed）
Execution protocol source: #96 Task E / ADR 0039

## 当前状态

- Gate I：**PASSED**（#83 原 Task 0）
- Gate T1：**PASSED**（E-048）
- Gate T2：旧 E-039 control 在 canonical cadence 下 **FAILED**（E-049）
- 当前：**BLOCKED BY #105**
- Gate T3 与 formal λ sweep：尚未执行

本 Issue 接管原 #83 中“canonical transfer → positive control → λ coarse sweep”的工作；#83 后续只保留候选多 seed confirmation 与 final A0/A1/A2 matched evaluation。

## 研究目标

在 canonical 0.5 s train/eval contract 下回答：

1. frozen calibrated reward representation 是否仍 identifiable；
2. #105 重新冻结的 canonical PPO control 是否能通过 effective-update transfer；
3. R0 vs λ=64 positive control 是否建立与 energy minimization 兼容的 objective→behavior direction transfer；
4. λ={1,2,4,8} 是否产生可重复、可解释的 dose-response；
5. 若成立，最多筛出 1–2 个候选交给 #83；若不成立，在对应 gate 明确收口。

最终不按 training return 选择 λ。

## 冻结研究 contract

除非在恢复 T2 前显式修改并重新冻结：

- frozen Diffusion Planner checkpoint / DDIM5；
- Exploration Policy architecture 与同一 initial checkpoint；
- training scenario family / map seeds；
- held-out S/SC map seeds 16–23；
- deterministic Beta mean 为正式主评测；
- calibrated Progress / Comfort constants 来自 E-034；
- calibrated efficiency-band energy representation / thresholds 来自 E-038；
- canonical execution = 5 × 0.1 s = 0.5 s；
- evaluation 使用 execution-trace fuel proxy；
- λ 主效应阶段只允许改变 energy weight λ；
- PPO control 由 #105 独立冻结，不在本 Issue 中重新搜索。

### Gate I — implementation/protocol gate【PASSED】

已完成：

- [x] λ={1,2,4,8} 可由正式 config/protocol compose；
- [x] arms 之间除 profile identity / weights.energy 外无 objective 差异；
- [x] λ=0 使用 calibrated R0；
- [x] resolved config 保存 λ、reward profile、canonical cadence、seeds；
- [x] comparison guard / reward rescore parity / tests 已完成；
- [x] 正式实验 commit / local test provenance 已登记。

### Gate T1 — objective identifiability transfer【PASSED, E-048】

canonical k=5 fixed source batch 下：

- endpoint actor-head cosine = 0.910851；
- normalized-advantage RMSE = 0.664142；
- sign-flip = 18.75%；
- λ16/64/256 分别达到 endpoint angular separation 的 58.79% / 85.29% / 95.89%。

因此 calibrated representation 已 transfer 到 canonical cadence。

### 历史 Gate T2【FAILED, E-049】

旧 E-039 control 在 canonical k=5 下：

- median seeded-MC post-update KL = 3.258e-7 < 1e-6；
- ratio change = 1.097e-3；
- deterministic Beta-mean probe RMS = 0.02205；
- c2–c6 通过；
- c7 因 c1 未过而未执行。

该历史失败不再在本 Issue 内归因或重搜；归 #105。

## Task 1 — canonical PPO re-entry / Gate T2

#105 完成后，把以下内容写回本 Issue：

- frozen canonical PPO resolved config；
- search / selection rule；
- analytic Beta KL 实现与 validation；
- attribution conclusion；
- provenance / experiment records。

然后使用 calibrated R0、canonical k=5、冻结 initial policy 与 matched protocol 重新执行 effective-update transfer。

### Gate T2 — re-entry

必须全部满足：

- [ ] non-trivial post-update KL，达到预注册 lower floor；
- [ ] >=90% updates KL 在 target / stability bound 内，无持续 runaway；
- [ ] policy ratio 有可测变化；
- [ ] deterministic Beta-mean distribution shift 非平凡；
- [ ] analytic Beta KL 与 MC/k3 对 update strength 给出一致数量级判断；
- [ ] 无 NaN / Inf / invalid action / Beta boundary collapse；
- [ ] collision / OOR / episode-length 无 pathological collapse；
- [ ] c7 held-out 非安全行为指标实际执行，并超过预注册 matched evaluation noise bound。

若 T2 失败：停止本 Issue 后续任务，不进入 positive control 或 λ sweep；将失败层级明确记录并回报 #80。

## Task 2 — canonical positive-control bridge / Gate T3

仅 T2 通过后执行：

~~~text
calibrated R0
vs
calibrated-band Rstress(λ=64)

2 matched training seeds
same initial policy
same update budget
same held-out matrix
canonical k=5
~~~

必须报告 guidance distribution、speed / progress / safety、total proxy energy 与 energy intensity，以及 executed behavior 与 canonical temporal mapping。

### Gate T3

- [ ] guidance separation 在两 seed 可重复；
- [ ] 至少一个闭环行为指标 separation 超过预注册 noise；
- [ ] energy-heavy arm 的 energy-intensity direction 与 energy minimization 兼容；
- [ ] direction 两个 training seeds 一致；
- [ ] 不能由 collision / OOR / wrong-direction / stopped / distance collapse 解释；
- [ ] executed behavior 可由 canonical k=5 planner/execution mapping 解释。

若 T3 失败：记录“canonical execution 下仍未建立可用 objective→behavior direction transfer”，停止普通 λ sweep。

## Task 3 — λ={0,1,2,4,8} formal dose-response / coarse sweep

仅 Gate I + T1 + T2 + T3 全通过后执行：

~~~text
λ = {0,1,2,4,8}
training seed = one frozen seed
updates = 50
checkpoint = 0 / 25 / 50
~~~

所有 arms 固定：

- same initial policy；
- same scenario/map seeds；
- same planner noise / policy-action RNG policy；
- same transition/update budget；
- same PPO config；
- same canonical execution；
- same held-out evaluation；
- same checkpoint cadence。

### 必须报告

Objective / PPO：

- normalized advantage / actor-gradient difference；
- analytic + seeded-MC KL；
- ratio / clip fraction；
- gradient norm / parameter delta；
- Beta mean / variance / concentration / boundary；
- guidance mean/std。

Behavior / task：

- speed profile / mean speed；
- route progress / completion / distance；
- stopped fraction；
- collision / OOR / wrong-direction / termination。

Energy：

- total execution-trace proxy energy；
- energy distance；
- primary energy intensity (mL/km)。

## Gate D — dose-response / candidate nomination

进入 #83 前至少要求：

- [ ] 至少两个非零 λ 相对 R0 出现超过预注册 noise 的 objective/guidance 或 behavior effect；
- [ ] λ 增大时 objective-pressure → guidance → executed behavior 方向总体可解释；
- [ ] energy-intensity 变化不由少走 / 停车 / 失败系统性混淆；
- [ ] 无 Beta boundary / KL runaway / invalid trajectory；
- [ ] 非单调行为若存在，有明确 progress/speed/safety trade-off 或 planner mapping 解释；
- [ ] 最多筛出 1–2 个候选，且筛选规则不使用单一 training return / 偶然 endpoint。

若所有普通 λ 均无 measurable effect，记录：

~~~text
no measurable effect after
identifiability + canonical-PPO transfer + positive-control gates
~~~

并结束本 Issue，不进入 #83 的候选确认。

## 输出 / 完成判据

本 Issue 结束时只能有两种合法结果：

### A. Candidate path

- [ ] T2/T3/D 全通过；
- [ ] 明确 1–2 个 λ candidates；
- [ ] 每个候选给出完整 provenance、50-update matched results 与选择理由；
- [ ] 将候选与冻结 contract handoff 给 #83。

### B. Stop path

- [ ] 明确停止在 T2、T3 或 D；
- [ ] 报告该 gate 的失败证据与可支持结论；
- [ ] 不伪造或补跑后续阶段；
- [ ] #83 标记为 not reached / blocked，不把缺失结果解释为 negative result。

## 非目标

本 Issue 不做：

- cadence-PPO attribution / hyperparameter re-freeze（归 #105）；
- candidates × 3 seeds × 100 updates；
- final A0/A1/A2 matched evaluation；
- reward representation、execution horizon、PPO 超参与 λ 的联合搜索。
